buildSrc/src/main/groovy/org/elasticsearch/gradle/test/DistroTestPlugin.java

```diff
@@ -65,6 +70,40 @@
 import static org.elasticsearch.gradle.vagrant.VagrantMachine.convertWindowsPath;
 
 public class DistroTestPlugin implements Plugin<Project> {
+    private static final Logger logger = Logging.getLogger(DistroTestPlugin.class);
+    /**
+     * This plugin generates test tasks for the various VMs in :qa:os. The
+     * Docker packaging tests are run on all of them, apart from those
+     * listed here. These values correspond to the sub-project names, which
+     * correspond to the directory names.
+     */
+    private static final List<String> DOCKER_VM_EXCLUDE_LIST = List.of(
```

Comment:
I'd prefer we not hard-code this kind of thing and instead have the projects themselves _declare_ this kind of thing.
Basically, the project should tell the plugin about this limitation, not the other way around.

I'd do something like:
1. Register an extension (or extra property) on each subproject to allow it to declare it supports Docker packaging tests.
2. Disable the docker packaging tests tasks via an `onlyIf` condition that looks for that extension property value.